### PR TITLE
Add supervisor role for CVHO and STS

### PR DESCRIPTION
**Pull request by timcheadle**
*Story:* https://www.pivotaltracker.com/story/show/158967518

Add user roles and authorization (using [Pundit](https://github.com/varvet/pundit)) to allow CVHOs and STSs to take certain actions in the backend UI

### CODE REVIEW

File: app/models/admin.rb [hunk 1]
```diff
@@ -1,4 +1,11 @@
 class Admin < ApplicationRecord
   devise :database_authenticatable, :lockable, :recoverable, :rememberable,
          :timeoutable, :trackable, :validatable
+
+  enum role: [
+    :owner,
```
Comment (govindkrjoshi) [inline app/models/admin.rb]: The name owner is ambiguous.

File: app/views/admin/facilities/index.html.erb [hunk 2]
```diff
@@ -28,13 +28,17 @@
         <td><%= facility.country %></td>
         <td><%= facility.pin %></td>
         <td><%= facility.facility_type %></td>
-        <td><%= link_to 'Edit', edit_admin_facility_path(facility) %></td>
-        <td><%= link_to 'Destroy', [:admin, facility], method: :delete, data: { confirm: 'Are you sure?' } %></td>
+        <% if policy(facility).edit? %>
+          <td><%= link_to 'Edit', edit_admin_facility_path(facility) %></td>
+          <td><%= link_to 'Destroy', [:admin, facility], method: :delete, data: { confirm: 'Are you sure?' } %></td>
```
Comment (govindkrjoshi) [inline app/views/admin/facilities/index.html.erb]: We should check if the user has destroy policy as a separate condition

**Comment by govindkrjoshi** [comment]
Should check for authorization for update and delete in the controllers as well?
The PR only shows edit or delete button if the user has permissions, but the controller should also check for permissions before allowing the actions.

File: app/models/admin.rb (hunk 1 above)
Comment (timcheadle) [reply app/models/admin.rb]: We both agree, but are unsure what to call it. We'll keep it as `:owner` for now and change in the future.

File: app/views/admin/facilities/index.html.erb (hunk 2 above)
Comment (timcheadle) [reply app/views/admin/facilities/index.html.erb]: 👍

**Pull request merged**